command/show.go

```diff
@@ -145,7 +145,30 @@ func (c *ShowCommand) Run(args []string) int {
 	if plan != nil {
 		if jsonOutput == true {
 			config := ctx.Config()
-			jsonPlan, err := jsonplan.Marshal(config, plan, stateFile, schemas)
+
+			var err error
+			var jsonPlan []byte
+
+			// If there is no prior state, we have all the schemas needed.
+			if stateFile == nil {
+				jsonPlan, err = jsonplan.Marshal(config, plan, stateFile, schemas, nil)
+			} else {
+				// If there is state, we need the state-specific schemas, which
+				// may differ from the schemas loaded from the plan.
+				// This occurs if there is a data_source in the state that was
+				// removed from the configuration, because terraform core does
+				// not need to load the schema to remove a data source.
+				opReq.PlanFile = nil
```

Comment:
I want to draw your attention to this line, which feels _cheesy_ and I'm concerned I'm overlooking either a problem this might cause later on, or a better method of extracting the schemas from the state.